dali/base/dadfs.cpp

```diff
@@ -2496,7 +2498,7 @@ class CDistributedFileIterator: public CDistributedFileIteratorBase<IDistributed
     bool set()
     {
         while (isValid()) {
-            cur.setown(parent->lookup(queryName(),udesc));
+            cur.setown(parent->lookup(queryName(),udesc, false, false, false, nullptr, true));
```

Comment:
CDistributedFileIterator comment shows that it is used to list files in directory -nothing there to read contents.  Should be ok?

Reply:
the iterator returns IDistributeFile's so any code that uses a IDistributedFileIterator - could use one and get a IDistributeFile and read content.

Whether there's any paths to that, through ECL code, I don't know, but doesn't feel like it should always default to give unsigned access.
e.g. if any engines uses one of these iterators to get files, it will bypass the signedness check.

The solution may be to pass in the isprivileged flag when creating one of these and ensure all uses pass in whether they are trusted or not.